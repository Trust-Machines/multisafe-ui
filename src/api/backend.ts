import {NETWORK} from '@trustmachines/multisafe-contracts';
import {DATA_BACKENDS} from '../constants';

// TODO: Use backend api for all (possible) data read operations
// TODO: Refactor SafeB to Safe

export type Status = 'mempool' | 'microblock' | 'anchor' | 'failed';

export interface SafeB {
  address: string,
  sender: string,
  tx_hash: string,
  threshold: number,
  version: string,
  nonce: number,
  owners: string[],
  time: number,
  status: Status,
  balance: string
}

export const getSafes = (network: NETWORK, owner: string): Promise<SafeB[]> => {
  return fetch(`${DATA_BACKENDS[network]}/safes/by_owner/${owner}`).then(r => r.json());
}
