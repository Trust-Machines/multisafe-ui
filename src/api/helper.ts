import {ClarityAbi} from '@stacks/transactions';
import {NETWORK} from '@trustmachines/multisafe-contracts';

export const validateSafeAbi = (abi: ClarityAbi) => {
    const publicFns = [
        'add-owner',
        'remove-owner',
        'set-threshold',
        'submit',
        'confirm',
        'revoke'
    ]

    for (let p of publicFns) {
        if (!abi.functions.find(x => x.name === p && x.access === 'public')) {
            return false;
        }
    }

    return true;
}

export const makeTxUrl = (txId: string, network: NETWORK) => {
    return `https://explorer.stacks.co/txid/${txId}?chain=${network}`;
}