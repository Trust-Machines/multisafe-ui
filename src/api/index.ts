import {StacksNetwork} from '@stacks/network';
import {
    callReadOnlyFunction,
    ClarityValue,
    cvToJSON,
    cvToValue,
    listCV,
    uintCV
} from '@stacks/transactions';
import {SafeTransaction} from '../store/safe';
import ftList from '../constants/ft-list';
import {FTAsset} from '../types';
import {TX_PER_PAGE} from '../constants';

export const getAccountMemPool = (network: StacksNetwork, account: string): Promise<{
    tx_id: string,
    contract_call?: {
        contract_id: string,
        function_name: string,
        function_args: {
            name: string
            repr: string
            type: string
        } []
    }
}[]> => {
    return fetch(`${network.coreApiUrl}/extended/v1/tx/mempool?sender_address=${account}`).then(r => r.json()).then(r => r.results);
}

export const getBnsName = (network: StacksNetwork, address: string): Promise<string | null> => {
    return fetch(`${network.coreApiUrl}/v1/addresses/stacks/${address}`).then(r => r.json()).then(r => {
        if (r.names.length > 0) {
            return r.names[0];
        }
        return null;
    })
}

export const resolveBnsName = (network: StacksNetwork, name: string): Promise<string | null> => {
    return fetch(`${network.coreApiUrl}/v1/names/${name}`).then(r => r.json()).then(r => {
        if (r.error) {
            return null;
        }
        return r.address;
    });
}

export const getContractInfo = (network: StacksNetwork, address: string, name: string): Promise<{ tx_id: string } | null> => {
    const contractId = `${address}.${name}`;
    return fetch(`${network.coreApiUrl}/extended/v1/contract/${contractId}?unanchored=true`).then(r => r.json()).then(r => {
        if (r.tx_id) {
            return r;
        }

        return null;
    });
}

export interface AddressBalance {
    stx: {
        balance: string
    },
    fungible_tokens: Record<string, { balance: string }>
}

export const getContractBalances = (network: StacksNetwork, address: string): Promise<AddressBalance> => {
    const [contractAddress, contractName] = address.split('.')
    return fetch(`${network.coreApiUrl}/extended/v1/address/${contractAddress}.${contractName}/balances`).then(r => r.json())
}

export const callReadOnly = (network: StacksNetwork, path: string, functionArgs: ClarityValue[], senderAddress: string) => {
    const [contractAddress, contractName, functionName] = path.split('.')
    return callReadOnlyFunction({
        contractAddress,
        contractName,
        functionName,
        functionArgs,
        senderAddress,
        network
    })
}

export interface SafeInfo {
    version: string,
    owners: string[],
    threshold: number,
    nonce: number
}

export const getSafeInfo = (network: StacksNetwork, safe: string, senderAddress: string): Promise<SafeInfo> => {
    return callReadOnly(network, `${safe}.get-info`, [], senderAddress).then(r => {
        const js = cvToJSON(r);
        return {
            version: js.value.version.value,
            owners: js.value.owners.value.map((x: any) => x.value),
            threshold: Number(js.value.threshold.value),
            nonce: Number(js.value.nonce.value),
        }
    });
}

export const getSafeTransactions = (network: StacksNetwork, safe: string, nonce: number, senderAddress: string): Promise<SafeTransaction[]> => {
    const txIds: number[] = [];
    for (let x = nonce - 1; x >= 0; x--) {
        txIds.push(x);

        if (txIds.length === TX_PER_PAGE) {
            break;
        }
    }
    const args = [listCV([...txIds.map(x => uintCV(x))])];
    return callReadOnly(network, `${safe}.get-transactions`, args, senderAddress).then(r => {
        return cvToJSON(r).value.map((x: any, i: number) => {

            return {
                id: txIds[i],
                confirmations: x.value.confirmations.value.map((c: any) => c.value),
                confirmed: x.value.confirmed.value,
                executor: x.value.executor.value,
                threshold: Number(x.value.threshold.value),
                paramFt: x.value['param-ft'].value,
                paramNft: x.value['param-nft'].value,
                paramP: x.value['param-p'].value ? x.value['param-p'].value.value : null,
                paramU: x.value['param-u'].value ? x.value['param-u'].value.value : null,
                paramB: x.value['param-b'].value ? x.value['param-b'].value.value : null,
            }
        });
    });
}

export const getFTInfo = async (network: StacksNetwork, address: string, senderAddress: string): Promise<FTAsset> => {
    const inList = ftList[network.isMainnet() ? 'mainnet' : 'testnet'].find(x => x.address === address);
    if (inList) {
        return inList;
    }

    return Promise.all([
        callReadOnly(network, `${address}.get-name`, [], senderAddress),
        callReadOnly(network, `${address}.get-symbol`, [], senderAddress),
        callReadOnly(network, `${address}.get-decimals`, [], senderAddress)
    ]).then(r => {
        return {
            address,
            name: cvToValue(r[0]).value,
            symbol: cvToValue(r[1]).value,
            decimals: Number(cvToValue(r[2]).value)
        }
    })
}