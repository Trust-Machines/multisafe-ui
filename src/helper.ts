import {ClarityAbi} from '@stacks/transactions';
import {NETWORK} from '@trustmachines/multisafe-contracts';
import BigNumber from 'bignumber.js';

export const validateSafeAbi = (abi: ClarityAbi) => {
    const publicFns = [
        'add-owner',
        'remove-owner',
        'set-min-confirmation',
        'submit',
        'confirm',
        'revoke'
    ]

    for (let p of publicFns) {
        if (!abi.functions.find(x => x.name === p && x.access === 'public')) {
            return false;
        }
    }

    return true;
}

export const makeTxUrl = (txId: string, network: NETWORK) => {
    return `https://explorer.stacks.co/txid/${txId}?chain=${network}`;
}

export const formatUnits = (value: BigNumber | string, decimals: number): BigNumber => {
    const bn = typeof value === 'string' ? new BigNumber(value) : value;
    return bn.dividedBy(10 ** decimals);
}

export const parseUnits = (value: string, decimals: number): BigNumber => {
    return new BigNumber(value).multipliedBy(10 ** decimals);
}