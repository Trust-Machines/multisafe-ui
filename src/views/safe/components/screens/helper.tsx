import React from 'react';
import Typography from '@mui/material/Typography';
import {Box, useTheme} from '@mui/material';

const ScreenHeader = (props: { title: string, icon: JSX.Element, children: JSX.Element }) => {
    const theme = useTheme();
    const icon = React.cloneElement(props.icon, {
        sx: {
            marginRight: '4px',
            color: theme.palette.primary.main
        }
    })
    return <Box sx={{
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'space-between',
        mb: '20px'
    }}>
        <Typography variant="h6" sx={{display: 'flex', alignItems: 'center',}}>
            {icon} {props.title}
        </Typography>
        {props.children}
    </Box>
}

export default ScreenHeader;