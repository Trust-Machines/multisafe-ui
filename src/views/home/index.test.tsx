import {renderWithRouter} from '../../helper/test-helper';
import useSafes from '../../hooks/use-safes';
import useAddress from '../../hooks/use-address';
import {SafeB} from '../../api/backend';
import Home from './index';


jest.mock('../../hooks/use-safes');
jest.mock('../../hooks/use-address');

test('1 Default render', () => {
    (useSafes as jest.Mock).mockReturnValue([{loading: false, list: []}]);
    const view = renderWithRouter(<Home/>);
    expect(view.container).toMatchSnapshot()
});


test('2 Wallet connected but no safe', () => {
    (useSafes as jest.Mock).mockReturnValue([{loading: false, list: []}]);
    (useAddress as jest.Mock).mockReturnValue('SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7')
    const view = renderWithRouter(<Home/>);
    expect(view.container).toMatchSnapshot()
});

test('3 Wallet connected and the user has wallets', () => {
    const safes: SafeB[] = [
        {
            address: 'SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7.safe-2',
            tx_hash: '0xd4563e48f835bd4df50f17861cf4550d9718cd06b2f1c4851e07c7670fc2ad47',
            sender: 'SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7',
            threshold: 2,
            version: '0.0.2.alpha',
            nonce: 1,
            owners: ['SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7', 'SP3T23YN6MBF44YNV910FD8JNMN1NZYGKG3MMZ73X'],
            time: 1658306070,
            status: 'anchor',
            balance: '10000'
        },
        {
            address: 'SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7.my-safe-19',
            tx_hash: '0xca84e1aebec1c0933c34bb1a2628b6b5cc69a6a9ebed777b3b5b10d8227411bf',
            sender: 'SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7',
            threshold: 2,
            version: '0.0.5.beta',
            nonce: 3,
            owners: ['SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7', 'SP2DXHX9Q844EBT80DYJXFWXJKCJ5FFAX50CQQAWN', 'SP2N7SK0W83NJSZHFH8HH31ZT3DXJG7NFE5VYT9SJ', 'SP3T23YN6MBF44YNV910FD8JNMN1NZYGKG3MMZ73X'],
            time: 1659454356,
            status: 'mempool',
            balance: '0'
        }
    ];

    (useSafes as jest.Mock).mockReturnValue([{loading: false, list: safes}]);
    (useAddress as jest.Mock).mockReturnValue('SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7')
    const view = renderWithRouter(<Home/>);
    expect(view.container).toMatchSnapshot()
});
