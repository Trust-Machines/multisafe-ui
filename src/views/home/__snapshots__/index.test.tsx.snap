// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 Default render 1`] = `
<div>
  <div
    class="MuiBox-root css-hoayk1"
  >
    <div
      class="MuiBox-root css-azxdeb"
    >
      <div
        class="MuiBox-root css-1qmgq4"
      >
        <div
          class="MuiBox-root css-wxdvxx"
        >
          <img
            alt="Logo"
            src="/logo.png"
          />
        </div>
        <div
          class="MuiBox-root css-1dx8jfp"
        >
          MultiSafe
        </div>
      </div>
    </div>
    <div
      class="flex-grow MuiBox-root css-0"
    />
    <div
      class="MuiBox-root css-ls0mhy"
    >
      <div
        class="MuiBox-root css-7lmw8c"
      >
        <div
          class="MuiBox-root css-198xhax"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18wkfow-MuiSvgIcon-root"
            data-testid="AccountBalanceWalletIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 18v1c0 1.1-.9 2-2 2H5c-1.11 0-2-.9-2-2V5c0-1.1.89-2 2-2h14c1.1 0 2 .9 2 2v1h-9c-1.11 0-2 .9-2 2v8c0 1.1.89 2 2 2h9zm-9-2h10V8H12v8zm4-2.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
            />
          </svg>
          <div
            class="MuiBox-root css-x1k79c"
          >
            <span
              class="first-line"
            >
              Not connected
            </span>
            <span
              class="second-line"
            >
              Connect Wallet
            </span>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1n1j97o"
      >
        <div
          class="MuiBox-root css-1sfu4nm"
        >
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-filledPrimary css-8w4ikn-MuiButtonBase-root-MuiChip-root"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              mainnet
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1qoaa23"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DarkModeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9 9-4.03 9-9c0-.46-.04-.92-.1-1.36-.98 1.37-2.58 2.26-4.4 2.26-2.98 0-5.4-2.42-5.4-5.4 0-1.81.89-3.42 2.26-4.4-.44-.06-.9-.1-1.36-.1z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-pi3xhg"
  >
    <div
      class="MuiBox-root css-8atqhb"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1pxvxmw-MuiTypography-root"
      >
        Welcome to MultiSafe
      </h4>
      <h6
        class="MuiTypography-root MuiTypography-h6 css-7vlfn4-MuiTypography-root"
      >
        MultiSafe is the most secure platform for storing your STX digital assets.
      </h6>
      <div
        class="MuiBox-root css-rabsv0"
      >
        <div
          class="MuiBox-root css-1fmbk1l"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
          >
            Don't have a safe?
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1ckb2mn-MuiTypography-root"
          >
            Create a safe with multiple owners
          </p>
          <div
            class="MuiBox-root css-0"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-2k1wi7-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-15ikgma-MuiSvgIcon-root"
                data-testid="AddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
               
              Create New Safe
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiBox-root css-ke8wp"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
          >
            Load Existing Safe
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1ckb2mn-MuiTypography-root"
          >
            Easily load your Safe using your Safe address
          </p>
          <div
            class="MuiBox-root css-0"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-rqrbu2-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-15ikgma-MuiSvgIcon-root"
                data-testid="RotateRightIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.55 5.55 11 1v3.07C7.06 4.56 4 7.92 4 12s3.05 7.44 7 7.93v-2.02c-2.84-.48-5-2.94-5-5.91s2.16-5.43 5-5.91V10l4.55-4.45zM19.93 11c-.17-1.39-.72-2.73-1.62-3.89l-1.42 1.42c.54.75.88 1.6 1.02 2.47h2.02zM13 17.9v2.02c1.39-.17 2.74-.71 3.9-1.61l-1.44-1.44c-.75.54-1.59.89-2.46 1.03zm3.89-2.42 1.42 1.41c.9-1.16 1.45-2.5 1.62-3.89h-2.02c-.14.87-.48 1.72-1.02 2.48z"
                />
              </svg>
              Load
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`2 Wallet connected but no safe 1`] = `
<div>
  <div
    class="MuiBox-root css-hoayk1"
  >
    <div
      class="MuiBox-root css-azxdeb"
    >
      <div
        class="MuiBox-root css-1qmgq4"
      >
        <div
          class="MuiBox-root css-wxdvxx"
        >
          <img
            alt="Logo"
            src="/logo.png"
          />
        </div>
        <div
          class="MuiBox-root css-1dx8jfp"
        >
          MultiSafe
        </div>
      </div>
    </div>
    <div
      class="flex-grow MuiBox-root css-0"
    />
    <div
      class="MuiBox-root css-ls0mhy"
    >
      <div
        class="MuiBox-root css-7lmw8c"
      >
        <div
          class="MuiBox-root css-198xhax"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18wkfow-MuiSvgIcon-root"
            data-testid="AccountBalanceWalletIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 18v1c0 1.1-.9 2-2 2H5c-1.11 0-2-.9-2-2V5c0-1.1.89-2 2-2h14c1.1 0 2 .9 2 2v1h-9c-1.11 0-2 .9-2 2v8c0 1.1.89 2 2 2h9zm-9-2h10V8H12v8zm4-2.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
            />
          </svg>
          <div
            class="MuiBox-root css-x1k79c"
          >
            <span
              class="first-line"
            >
              SP3XD...KAEK7
            </span>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1n1j97o"
      >
        <div
          class="MuiBox-root css-1sfu4nm"
        >
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-filledPrimary css-8w4ikn-MuiButtonBase-root-MuiChip-root"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              mainnet
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1qoaa23"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DarkModeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9 9-4.03 9-9c0-.46-.04-.92-.1-1.36-.98 1.37-2.58 2.26-4.4 2.26-2.98 0-5.4-2.42-5.4-5.4 0-1.81.89-3.42 2.26-4.4-.44-.06-.9-.1-1.36-.1z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-pi3xhg"
  >
    <div
      class="MuiBox-root css-8atqhb"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1pxvxmw-MuiTypography-root"
      >
        Welcome to MultiSafe
      </h4>
      <h6
        class="MuiTypography-root MuiTypography-h6 css-7vlfn4-MuiTypography-root"
      >
        MultiSafe is the most secure platform for storing your STX digital assets.
      </h6>
      <div
        class="MuiBox-root css-rabsv0"
      >
        <div
          class="MuiBox-root css-1fmbk1l"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
          >
            Don't have a safe?
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1ckb2mn-MuiTypography-root"
          >
            Create a safe with multiple owners
          </p>
          <div
            class="MuiBox-root css-0"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-2k1wi7-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-15ikgma-MuiSvgIcon-root"
                data-testid="AddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
               
              Create New Safe
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiBox-root css-ke8wp"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
          >
            Load Existing Safe
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1ckb2mn-MuiTypography-root"
          >
            Easily load your Safe using your Safe address
          </p>
          <div
            class="MuiBox-root css-0"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-rqrbu2-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-15ikgma-MuiSvgIcon-root"
                data-testid="RotateRightIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.55 5.55 11 1v3.07C7.06 4.56 4 7.92 4 12s3.05 7.44 7 7.93v-2.02c-2.84-.48-5-2.94-5-5.91s2.16-5.43 5-5.91V10l4.55-4.45zM19.93 11c-.17-1.39-.72-2.73-1.62-3.89l-1.42 1.42c.54.75.88 1.6 1.02 2.47h2.02zM13 17.9v2.02c1.39-.17 2.74-.71 3.9-1.61l-1.44-1.44c-.75.54-1.59.89-2.46 1.03zm3.89-2.42 1.42 1.41c.9-1.16 1.45-2.5 1.62-3.89h-2.02c-.14.87-.48 1.72-1.02 2.48z"
                />
              </svg>
              Load
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`3 Wallet connected and the user has wallets 1`] = `
<div>
  <div
    class="MuiBox-root css-hoayk1"
  >
    <div
      class="MuiBox-root css-azxdeb"
    >
      <div
        class="MuiBox-root css-1qmgq4"
      >
        <div
          class="MuiBox-root css-wxdvxx"
        >
          <img
            alt="Logo"
            src="/logo.png"
          />
        </div>
        <div
          class="MuiBox-root css-1dx8jfp"
        >
          MultiSafe
        </div>
      </div>
    </div>
    <div
      class="flex-grow MuiBox-root css-0"
    />
    <div
      class="MuiBox-root css-ls0mhy"
    >
      <div
        class="MuiBox-root css-7lmw8c"
      >
        <div
          class="MuiBox-root css-198xhax"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18wkfow-MuiSvgIcon-root"
            data-testid="AccountBalanceWalletIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 18v1c0 1.1-.9 2-2 2H5c-1.11 0-2-.9-2-2V5c0-1.1.89-2 2-2h14c1.1 0 2 .9 2 2v1h-9c-1.11 0-2 .9-2 2v8c0 1.1.89 2 2 2h9zm-9-2h10V8H12v8zm4-2.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
            />
          </svg>
          <div
            class="MuiBox-root css-x1k79c"
          >
            <span
              class="first-line"
            >
              SP3XD...KAEK7
            </span>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1n1j97o"
      >
        <div
          class="MuiBox-root css-1sfu4nm"
        >
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-filledPrimary css-8w4ikn-MuiButtonBase-root-MuiChip-root"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              mainnet
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1qoaa23"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DarkModeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9 9-4.03 9-9c0-.46-.04-.92-.1-1.36-.98 1.37-2.58 2.26-4.4 2.26-2.98 0-5.4-2.42-5.4-5.4 0-1.81.89-3.42 2.26-4.4-.44-.06-.9-.1-1.36-.1z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-pi3xhg"
  >
    <div
      class="MuiBox-root css-8atqhb"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1pxvxmw-MuiTypography-root"
      >
        Welcome to MultiSafe
      </h4>
      <h6
        class="MuiTypography-root MuiTypography-h6 css-7vlfn4-MuiTypography-root"
      >
        MultiSafe is the most secure platform for storing your STX digital assets.
      </h6>
      <div
        class="MuiBox-root css-rabsv0"
      >
        <div
          class="MuiBox-root css-ke8wp"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Your Safes
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-wxj4sl-MuiTypography-root"
          >
            List of Safes created by you or have you as one of its owners.
          </p>
          <div
            class="MuiBox-root css-30wj9f"
          >
            <div
              class="MuiBox-root css-espiy2"
            >
              my-safe-19
            </div>
            <div
              class="MuiBox-root css-1lh8mp8"
            >
              SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7
            </div>
            <div
              class="MuiBox-root css-lpjzzz"
            >
              <span
                class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
                role="progressbar"
                style="width: 0.6rem; height: 0.6rem;"
              >
                <svg
                  class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <circle
                    class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                    cx="44"
                    cy="44"
                    fill="none"
                    r="20.2"
                    stroke-width="3.6"
                  />
                </svg>
              </span>
               
              being deployed
            </div>
          </div>
          <div
            class="MuiBox-root css-1mcng6u"
          >
            <div
              class="MuiBox-root css-espiy2"
            >
              safe-2
            </div>
            <div
              class="MuiBox-root css-1lh8mp8"
            >
              SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7
            </div>
          </div>
          <div
            class="MuiBox-root css-qj0ggg"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall css-29k7bz-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-15ikgma-MuiSvgIcon-root"
                data-testid="AddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
               
              Create new Safe
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall css-1k23hlb-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-15ikgma-MuiSvgIcon-root"
                data-testid="RotateRightIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.55 5.55 11 1v3.07C7.06 4.56 4 7.92 4 12s3.05 7.44 7 7.93v-2.02c-2.84-.48-5-2.94-5-5.91s2.16-5.43 5-5.91V10l4.55-4.45zM19.93 11c-.17-1.39-.72-2.73-1.62-3.89l-1.42 1.42c.54.75.88 1.6 1.02 2.47h2.02zM13 17.9v2.02c1.39-.17 2.74-.71 3.9-1.61l-1.44-1.44c-.75.54-1.59.89-2.46 1.03zm3.89-2.42 1.42 1.41c.9-1.16 1.45-2.5 1.62-3.89h-2.02c-.14.87-.48 1.72-1.02 2.48z"
                />
              </svg>
               
              Load another safe 
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
